src/components/treemap.tsx

```diff
@@ -80,19 +80,48 @@ export default class Treemap extends React.Component<
       width={d.x1 - d.x0}
       height={d.y1 - d.y0}
     />
-  ));
+  ))
+
+    sizingProperties = d => {
+        const [ minValue, maxValue ] = this.props.fontSize;
+        const x = max([0, Math.floor(((d.x1 - d.x0) - 8))]);
+        const y = max([0, Math.floor(((d.y1 - d.y0) - 8))]);
+        const label = d.data.location_name;
+        const width = stringWidth(label, 12) / 12;
+        const direction = (y > x && (width * (minValue + 1)) > x) ? 'topBottom' : 'leftRight';
+        return { x, y, width, direction, minValue, maxValue };
+    }
+
+    fontSize = d => {
+        const { x, y, width, direction, minValue, maxValue } = this.sizingProperties(d);
+        let size;
+        if (direction === 'leftRight') { size = (y < (x / width)) ? y : (x / width); }
+        if (direction === 'topBottom') { size = (x < (y / width)) ? x : (y / width); }
+        if (size < minValue) { size = 0; }
+        if (size >= maxValue) { size = maxValue; }
+        return size;
+    }
+
+    fontDirection = d => {
+        const { direction } = this.sizingProperties(d);
+        const orientation = {
+            leftRight: `translate(3px, ${this.fontSize(d)}px) rotate(0)`,
+            topBottom: `translate(${this.fontSize(d) / 3}px, 3px) rotate(90deg)`,
+        };
+        return (direction === 'leftRight') ? orientation.leftRight : orientation.topBottom;
+    }
 
   renderText = (d) => ((
```

Comment:
I don't think it's something we need to do in this PR, but it'd probably good to break this `<text />` element into it's own wrapper module soon. Maybe `<DynamicText />` or something...dealer's choice on the name.

It'll make animating the component a bit more bite size as well (I think) when we get to that step.